### PR TITLE
UX: Icons: Remove IconCheck and fa-check

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

Removes the usage of both the `<IconCheck />` component as well as `fa-check`.

### Account Menu
CheckAccountMenu

### Network Menu
NetworkCheck

### Transaction Confirmed

TransactionConfirmed

### EditableLabel
Check

### Onboarding Create Password

CheckmarkCreatePassword

### Onboarding MetaMetrics
OnboardingMetametricsCheck

### Send Domain
SendDomain

### Settings Network
SettingsNetwork

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/account-menu/account-menu.component.js [hunk 1]
```diff
@@ -231,7 +230,7 @@ export default class AccountMenu extends Component {
         >
           <div className="account-menu__check-mark">
             {isSelected ? (
-              <IconCheck color="var(--color-success-default)" />
+              <Icon color={Color.successDefault} name={ICON_NAMES.CLOSE} />
```
Comment (georgewrmarshall) [inline ui/components/app/account-menu/account-menu.component.js:233]:
```suggestion
              <Icon color={Color.successDefault} name={ICON_NAMES.CHECK} />
```

File: ui/components/app/modals/metametrics-opt-in-modal/metametrics-opt-in-modal.component.js [hunk 2]
```diff
@@ -44,13 +45,13 @@ export default class MetaMetricsOptInModal extends Component {
 
               <div className="metametrics-opt-in__committments">
                 <div className="metametrics-opt-in__row">
-                  <i className="fa fa-check" />
+                  <Icon name={ICON_NAMES.CLOSE} />
```
Comment (georgewrmarshall) [inline ui/components/app/modals/metametrics-opt-in-modal/metametrics-opt-in-modal.component.js:48]:
```suggestion
                  <Icon name={ICON_NAMES.CHECK} />
```

File: ui/components/app/modals/metametrics-opt-in-modal/metametrics-opt-in-modal.component.js [hunk 3]
```diff
@@ -44,13 +45,13 @@ export default class MetaMetricsOptInModal extends Component {
...
                   <div className="metametrics-opt-in__row-description">
                     {t('metametricsCommitmentsAllowOptOut')}
                   </div>
                 </div>
                 <div className="metametrics-opt-in__row">
-                  <i className="fa fa-check" />
+                  <Icon name={ICON_NAMES.CLOSE} />
```
Comment (georgewrmarshall) [inline ui/components/app/modals/metametrics-opt-in-modal/metametrics-opt-in-modal.component.js:54]:
```suggestion
                  <Icon name={ICON_NAMES.CHECK} />
```

File: ui/components/app/modals/transaction-confirmed/transaction-confirmed.component.js [hunk 4]
```diff
@@ -28,9 +30,10 @@ export default class TransactionConfirmed extends PureComponent {
     return (
       <Modal onSubmit={this.handleSubmit} submitText={t('ok')}>
         <div className="transaction-confirmed__content">
-          <i
-            className="fa fa-check-circle fa-3x"
-            style={{ color: 'var(--color-success-default)' }}
+          <Icon
+            name={ICON_NAMES.CLOSE}
```
Comment (georgewrmarshall) [inline ui/components/app/modals/transaction-confirmed/transaction-confirmed.component.js:34]:
```suggestion
            name={ICON_NAMES.CHECK}
```

File: ui/components/ui/editable-label/editable-label.js [hunk 5]
```diff
@@ -62,7 +62,7 @@ export default class EditableLabel extends Component {
           className="editable-label__icon-button"
           onClick={() => this.handleSubmit(isValidAccountName)}
         >
-          <i className="fa fa-check editable-label__icon" />
+          <Icon name={ICON_NAMES.CLOSE} className="editable-label__icon" />
```
Comment (georgewrmarshall) [inline ui/components/ui/editable-label/editable-label.js:65]:
```suggestion
          <Icon name={ICON_NAMES.CHECK} className="editable-label__icon" />
```

File: ui/components/ui/form-field/form-field.stories.js [hunk 6]
```diff
@@ -69,7 +70,7 @@ export const FormFieldWithTitleDetail = (args) => {
         Click Me
       </button>
     ),
-    checkmark: <i className="fas fa-check" />,
+    checkmark: <Icon name={ICON_NAMES.CLOSE} />,
```
Comment (georgewrmarshall) [inline ui/components/ui/form-field/form-field.stories.js:73]:
```suggestion
    checkmark: <Icon name={ICON_NAMES.CHECK} />,
```

File: ui/pages/onboarding-flow/create-password/create-password.js [hunk 7]
```diff
@@ -220,7 +221,7 @@ export default function CreatePassword({
             titleDetail={
               isValid && (
                 <div className="create-password__form--checkmark">
-                  <i className="fas fa-check" />
+                  <Icon name={ICON_NAMES.CLOSE} />
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/create-password/create-password.js:224]:
```suggestion
                  <Icon name={ICON_NAMES.CHECK} />
```

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 8]
```diff
@@ -86,11 +88,11 @@ export default function OnboardingMetametrics() {
       </Typography>
       <ul>
         <li>
-          <i className="fa fa-check" />
+          <Icon name={ICON_NAMES.CLOSE} color={Color.successDefault} />
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:91]:
```suggestion
          <Icon name={ICON_NAMES.CHECK} color={Color.successDefault} />
```

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 9]
```diff
@@ -86,11 +88,11 @@ export default function OnboardingMetametrics() {
...
           {t('onboardingMetametricsAllowOptOut')}
         </li>
         <li>
-          <i className="fa fa-check" />
+          <Icon name={ICON_NAMES.CLOSE} color={Color.successDefault} />
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:95]:
```suggestion
          <Icon name={ICON_NAMES.CHECK} color={Color.successDefault} />
```

File: ui/pages/permissions-connect/redirect/permissions-redirect.component.js [hunk 10]
```diff
@@ -36,7 +41,11 @@ export default function PermissionsRedirect({ subjectMetadata }) {
             alignItems={AlignItems.center}
             justifyContent={JustifyContent.center}
           >
-            <i className="fa fa-check fa-lg permissions-redirect__check" />
+            <Icon
+              name={ICON_NAMES.CLOSE}
```
Comment (georgewrmarshall) [inline ui/pages/permissions-connect/redirect/permissions-redirect.component.js:45]:
```suggestion
              name={ICON_NAMES.CHECK}
```

File: ui/pages/send/send-content/add-recipient/domain-input.component.js [hunk 11]
```diff
@@ -104,9 +104,10 @@ export default class DomainInput extends Component {
           })}
         >
           {hasSelectedAddress ? (
-            <i
-              className="ens-input__wrapper__status-icon fa fa-check-circle"
-              style={{ color: 'var(--color-success-default)' }}
+            <Icon
+              className="ens-input__wrapper__status-icon"
+              name={ICON_NAMES.CLOSE}
```
Comment (georgewrmarshall) [inline ui/pages/send/send-content/add-recipient/domain-input.component.js:109]:
```suggestion
              name={ICON_NAMES.CHECK}
```

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js [hunk 12]
```diff
@@ -72,17 +73,9 @@ const NetworksListItem = ({
       }}
     >
       {isCurrentRpcTarget ? (
-        <IconCheck
-          className="networks-tab__content__icon-check"
-          color="var(--color-success-default)"
-          aria-label={t('active')}
-        />
+        <Icon name={ICON_NAMES.CLOSE} color={Color.successDefault} />
```
Comment (georgewrmarshall) [inline ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:76]:
```suggestion
        <Icon name={ICON_NAMES.CHECK} color={Color.successDefault} />
```

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js [hunk 13]
```diff
@@ -72,17 +73,9 @@ const NetworksListItem = ({
...
       ) : (
-        <IconCheck
-          className="networks-tab__content__icon-check"
-          color="transparent"
-          aria-hidden="true"
-        />
+        <Icon name={ICON_NAMES.CLOSE} color={Color.transparent} />
```
Comment (georgewrmarshall) [inline ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:78]:
```suggestion
        <Icon name={ICON_NAMES.CHECK} color={Color.transparent} />
```

**Review by georgewrmarshall — DISMISSED** [review]
Need to update `CLOSE` to `CHECK`

**Comment by darkwing** [comment]
*Sigh* I keep jumping between all these Icon PRs, I must've lost my mind.  Thank you @georgewrmarshall !

File: ui/components/app/account-menu/account-menu.component.js [hunk 14]
```diff
@@ -231,7 +230,7 @@ export default class AccountMenu extends Component {
         >
           <div className="account-menu__check-mark">
             {isSelected ? (
-              <IconCheck color="var(--color-success-default)" />
+              <Icon color={Color.successDefault} name={ICON_NAMES.CHECK} />
             ) : null}
```
Comment (darkwing) [inline ui/components/app/account-menu/account-menu.component.js:234]: CheckAccountMenu

File: ui/components/app/account-menu/account-menu.component.js [hunk 15]
```diff
@@ -231,7 +224,11 @@ export default class AccountMenu extends Component {
         >
           <div className="account-menu__check-mark">
             {isSelected ? (
-              <IconCheck color="var(--color-success-default)" />
+              <Icon
+                color={Color.successDefault}
```
Comment (DDDDDanica) [inline ui/components/app/account-menu/account-menu.component.js:228]: Can you use `IconColor` enum here?

File: ui/components/app/dropdowns/network-dropdown.js [hunk 16]
```diff
@@ -175,7 +179,7 @@ class NetworkDropdown extends Component {
           }}
         >
           {isCurrentRpcTarget ? (
-            <IconCheck color="var(--color-success-default)" />
+            <Icon name={ICON_NAMES.CHECK} color={Color.successDefault} />
```
Comment (DDDDDanica) [inline ui/components/app/dropdowns/network-dropdown.js:182]: Can you use `IconColor` enum here?

File: ui/components/app/dropdowns/network-dropdown.js [hunk 17]
```diff
@@ -243,7 +247,7 @@ class NetworkDropdown extends Component {
         style={DROP_DOWN_MENU_ITEM_STYLE}
       >
         {providerType === network ? (
-          <IconCheck color="var(--color-success-default)" />
+          <Icon name={ICON_NAMES.CHECK} color={Color.successDefault} />
```
Comment (DDDDDanica) [inline ui/components/app/dropdowns/network-dropdown.js:250]: Can you use `IconColor` enum here?

File: ui/components/app/modals/transaction-confirmed/transaction-confirmed.component.js [hunk 18]
```diff
@@ -28,9 +30,10 @@ export default class TransactionConfirmed extends PureComponent {
     return (
       <Modal onSubmit={this.handleSubmit} submitText={t('ok')}>
         <div className="transaction-confirmed__content">
-          <i
-            className="fa fa-check-circle fa-3x"
-            style={{ color: 'var(--color-success-default)' }}
+          <Icon
+            name={ICON_NAMES.CHECK}
+            color={Color.successDefault}
```
Comment (DDDDDanica) [inline ui/components/app/modals/transaction-confirmed/transaction-confirmed.component.js:35]: Can you use `IconColor` enum here?

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 19]
```diff
@@ -86,11 +88,19 @@ export default function OnboardingMetametrics() {
       </Typography>
       <ul>
         <li>
-          <i className="fa fa-check" />
+          <Icon
+            name={ICON_NAMES.CHECK}
+            color={Color.successDefault}
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:93]: Can you use `IconColor` enum here?

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js [hunk 20]
```diff
@@ -77,17 +76,9 @@ const NetworksListItem = ({
       }}
     >
       {isCurrentRpcTarget ? (
-        <IconCheck
-          className="networks-tab__content__icon-check"
-          color="var(--color-success-default)"
-          aria-label={t('active')}
-        />
+        <Icon name={ICON_NAMES.CHECK} color={Color.successDefault} />
```
Comment (DDDDDanica) [inline ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:79]: Can you use `IconColor` enum here?

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js [hunk 21]
```diff
@@ -77,17 +76,9 @@ const NetworksListItem = ({
...
       ) : (
-        <IconCheck
-          className="networks-tab__content__icon-check"
-          color="transparent"
-          aria-hidden="true"
-        />
+        <Icon name={ICON_NAMES.CHECK} color={Color.transparent} />
```
Comment (DDDDDanica) [inline ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:81]: Can you use `IconColor` enum here?

**Comment by DDDDDanica** [comment]
Hey @darkwing just wanna let you know typescript enum changes made in [this MR](https://github.com/MetaMask/metamask-extension/pull/17518) will give us better enums to use, for Icons specifically we can use `IconColor` which was exposed before, and now will define a better scope for colors available for Icon~

File: ui/components/ui/editable-label/editable-label.js [hunk 22]
```diff
@@ -58,12 +58,11 @@ export default class EditableLabel extends Component {
           })}
           autoFocus
         />
-        <button
+        <ButtonIcon
+          iconName={ICON_NAMES.CHECK}
```
Comment (georgewrmarshall) [inline ui/components/ui/editable-label/editable-label.js:62]: This will need an `ariaLabel` prop to pass accessibility

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Sorry, forgot I left a suggestion above ☝️

File: ui/components/app/modals/transaction-confirmed/transaction-confirmed.component.js (hunk 18 above)
Comment (DDDDDanica) [inline ui/components/app/modals/transaction-confirmed/transaction-confirmed.component.js:35]:
```suggestion
            color={IconColor.successDefault}
```

File: ui/pages/send/send-content/add-recipient/domain-input.component.js [hunk 23]
```diff
@@ -104,9 +104,10 @@ export default class DomainInput extends Component {
           })}
         >
           {hasSelectedAddress ? (
-            <i
-              className="ens-input__wrapper__status-icon fa fa-check-circle"
-              style={{ color: 'var(--color-success-default)' }}
+            <Icon
+              className="ens-input__wrapper__status-icon"
+              name={ICON_NAMES.CHECK}
+              color={Color.successDefault}
```
Comment (DDDDDanica) [inline ui/pages/send/send-content/add-recipient/domain-input.component.js:110]:
```suggestion
              color={IconColor.successDefault}
```

**Review by DDDDDanica — DISMISSED** [review]

**Review by georgewrmarshall — DISMISSED** [review]

**Review by georgewrmarshall — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**